charts/ingress-nginx/tests/controller-daemonset_test.yaml

```
@@ -95,3 +95,42 @@ tests:
               topologyKey: kubernetes.io/hostname
               maxSkew: 1
               whenUnsatisfiable: ScheduleAnyway
+
+  - it: should create a DaemonSet with affinity if `controller.affinity` is set
+    set:
+      controller.kind: DaemonSet
+      controller.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution:
```

```suggestion
      controller.affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
```

This should also work.

---

I considered doing it this way, at least that was my first way I did it without even thinking, then I saw that every field in all other tests doesn't have any nesting so I assumed you wanted it as compact as possible.